geonode_mapstore_client/client/js/utils/ResourceUtils.js

```diff
@@ -696,3 +696,18 @@ export const getResourceImageSource = (image) => {
 export const isDocumentExternalSource = (resource) => {
     return resource && resource.resource_type === ResourceTypes.DOCUMENT && resource.sourcetype === 'REMOTE';
 };
+
+export const getDownloadUrlInfo = (resource) => {
+    let downloadUrlInfo;
+    if (!isEmpty(resource?.download_urls)) {
+        const downloadData = resource.download_urls.length === 1
+            ? resource.download_urls[0]
+            : resource.download_urls.find((d) => d.default);
+        if (!isEmpty(downloadData)) {
+            downloadUrlInfo = { url: downloadData.url, ajaxSafe: downloadData.ajax_safe };
+        }
+    } else {
+        downloadUrlInfo = { url: resource?.href, ajaxSafe: false };
+    }
+    return downloadUrlInfo;
+};
```

Comment:
I would prefer to use a return structure instead to mutate the object downloadUrlInfo.
I think it's make it more clear what's happening when the download_urls is not available but we could discuss this.

please double check if it works, if you accept to use this code
```js
export const getDownloadUrlInfo = (resource) => {
    if (!isEmpty(resource?.download_urls)) {
        const downloadData = resource.download_urls.length === 1
            ? resource.download_urls[0]
            : resource.download_urls.find((d) => d.default);
        if (!isEmpty(downloadData)) {
            return { url: downloadData.url, ajaxSafe: downloadData.ajax_safe };
        }
    }
    return { url: resource?.href, ajaxSafe: false };
};
```

Reply:
Ok